blog/zero_trust_security.md

```diff
@@ -0,0 +1,59 @@
+
+The WASI extension to the specification allows modules to do a few more things that let them pretend to be real applications, like communicate with stdout/stderr, file I/O, etc. Even with this in place, the runtime itself is responsible for making the virtual file system available to the module, and can deny any attempt to utilize any file descriptor (including `stdout`).
+
+![capability security](/images/blogs/ztsec/zt2.png)
+
+### Capability Security
+Once we get above the basic runtime level (we also often call this the "engine" level), we have wasmCloud (which uses an engine/runtime internally) and our ability to securely and dynamically bind capabilities to actors. With wasmCloud, we use signed JSON Web Tokens ([JWT](https://jwt.io)) embedded directly in the module. This allows us to verify the permissions of a module without consulting a central authority that can be spoofed or compromised. This means that we can still make informed decisions about what an actor can or cannot do in offline or constrained environments or during network partition events.
+ 
+Actors are signed with identifiers for the capability contracts to which they have been granted access, like `HTTP Server`, `Message Subscriber`, or `Key Value Store`. These capabilities are abstract contracts, the implementations for which are linked at runtime. You can use mock, test, or lightweight implementations when going through your inner development loop on your workstation and then swap to different implementations in higher environments, all without having to rebuild or redeploy your actors.
+
+Because actors are cryptographically signed, we can also subject actors to policies that utilize metadata like the issuer of the module, its expiration date, its "not valid before" date, and much more. We can use policies like this to only allow actors from a certain set of trusted issuers, split the issuers allowed for production and dev/test, and any number of other additional security measures.
+
+### Cluster Security
+As we continue moving up the abstraction level from a single wasmCloud host, next we can look at a cluster of wasmCloud hosts known as a _lattice_. In a lattice, only those hosts that have been authorized to communicate with the cluster can do so, and invocations from invalid issuers or from spoofed senders or MITM (man-in-the-middle) attackers can all be rejected.
+
+To accomplish this, each host in a wasmCloud cluster generates its own unique identity key pair, but is also given a _cluster signing key_. This key is used to sign invocations to actors and providers elsewhere on the cluster. These remote hosts receiving those calls can then verify the issuer of that invocation, as well as whether the invocation's bytes have been tampered with.
```

Comment:
for `cluster signing key` this is a good link to the zero trust section in the docs